### ISSUE TITLE
Setup failed for custom integration 'alexa_media': Requirements for alexa_media not found: ['alexapy==1.29.4']

### ISSUE THREAD

**Issue by hareeshmu** [alandtse/alexa_media_player#2720]
**IMPORTANT: Please search the issues, including closed issues, and the [FAQ](https://github.com/alandtse/alexa_media_player/wiki/FAQ) before opening a new issue. The template is mandatory; failure to use it will result in issue closure.**

**Describe the bug**

Home Assistant Core: 2024.12.0

Setup failed for custom integration 'alexa_media': Requirements for alexa_media not found: ['alexapy==1.29.4']

Unable to install package alexapy==1.29.4: × No solution found when resolving dependencies: ╰─▶ Because alexapy==1.29.4 depends on aiofiles>=23.1.0,<24.0.0 and aiofiles>=24.1.0, we can conclude that alexapy==1.29.4 cannot be used. And because you require alexapy==1.29.4, we can conclude that your requirements are unsatisfiable.

**To Reproduce**

Upgrade to Home Assistant Core: 2024.12.0

**Expected behavior**

**Screenshots**

**System details**

- Home Assistant version: 2024.12.0
- alexa_media version (from `const.py` or HA startup log):
- alexapy version (from `pip show alexapy` in homeasssistant container or HA startup log): 1.29.4
- Is Amazon [2FA/2SV enabled](https://github.com/alandtse/alexa_media_player/wiki/Configuration#enable-two-factor-authentication-for-your-amazon-account)  (y/n):
- Amazon Domain:

**Debug Logs (alexa_media & alexapy)**
Please provide [logs](https://github.com/alandtse/alexa_media_player/wiki/FAQ#how-do-i-enable-debug-logging-for-the-component).

**Additional context**

**Comment by phxMurray** [comment]
same issue for me, guess that's why everyone says to avoid the .0 versions.  Tried rolling back to 4.13.8 and no change.

**Comment by vaibhav2912** [comment]
Same for me

**Comment by ptyers** [comment]
ditto here. upgraded to 2024.12 and alexa_media no longer able to be loaded. rolled back to earlier version from latest alexa version stll fails.

**Comment by fhahn28** [comment]
Same here

**Comment by Kiwistream** [comment]
Yip same here...

**Comment by naamah75** [comment]
Obviously same problem here...

**Comment by Rizzonelli** [comment]
Same here

**Comment by Uscha66** [comment]
same here

**Comment by Hannemann64** [comment]
same after update to H.A.C. 2024.12.0

**Comment by hareeshmu** [comment]
https://gitlab.com/keatontaylor/alexapy/-/issues/33

**Comment by Shaunpottie** [comment]
Same, Following

**Comment by ValentinBaier** [comment]
+1 following

I'm on a docker installation and it seems theres incompatabilites between aiofiles.
I've been able to isolate the problem by enabling a dedicated venv for this integration but it's more of a hotfix rather than a long-term solution.
Rolling back to 4.13.8 didnt seem to help.

**Comment by weapons** [comment]
Same here after the update to H.A.C. 2024.12.0

**Comment by qfina** [comment]
Spamming 'same here' is not useful, nor necessary.  Add something of value, or just subscribe to the issue.

**Comment by gluca** [comment]
Here the home assistant log error:
Registratore: homeassistant.util.package
Fonte: util/package.py:152
Prima occorrenza: 08:49:10 (3 occorrenze)
Ultimo accesso: 08:49:12

Unable to install package alexapy==1.29.4: × No solution found when resolving dependencies: ╰─▶ Because alexapy==1.29.4 depends on aiofiles>=23.1.0,<24.0.0 and aiofiles>=24.1.0, we can conclude that alexapy==1.29.4 cannot be used. And because you require alexapy==1.29.4, we can conclude that your requirements are unsatisfiable.

**Comment by eyal38** [comment]
same for me

**Comment by itmcgdb** [comment]
Same for me :(

**Comment by githubsharezone** [comment]
same for me ...

**Comment by Finest** [comment]
just use the damn subscribe button at the top right ... same for me or same here ... helps nobody ...

**Comment by HanyDaim** [comment]
Same here after updating to Home Assistant Core 2024.12.0, restored to previous version and everything reworking again. Waiting solution…

**Comment by falkzilm** [comment]
@ValentinBaier Could you may share a short snippet or how you did this venv change?
I usually do not work often with python, but introducing venv and setting python interpreter via shebang did not worked yet for me.

**Comment by abjwsm** [comment]
I have the same error post 2024.12.0
Logger: homeassistant.setup
Source: setup.py:269
First occurred: 4 December 2024 at 21:13:13 (1 occurrences)
Last logged: 4 December 2024 at 21:13:13

Setup failed for custom integration 'alexa_media': Requirements for alexa_media not found: ['alexapy==1.29.4'].

**Comment by Andyphillips27** [comment]
> Here the home assistant log error: Registratore: homeassistant.util.package Fonte: util/package.py:152 Prima occorrenza: 08:49:10 (3 occorrenze) Ultimo accesso: 08:49:12
> 
> Unable to install package alexapy==1.29.4: × No solution found when resolving dependencies: ╰─▶ Because alexapy==1.29.4 depends on aiofiles>=23.1.0,<24.0.0 and aiofiles>=24.1.0, we can conclude that alexapy==1.29.4 cannot be used. And because you require alexapy==1.29.4, we can conclude that your requirements are unsatisfiable.

Same issue after update this morning to 2024.12.0 :(

**Comment by Destroyer061090** [comment]
same issues with lastest HA version

**Comment by frankfrommelt** [comment]
I had been using 5.0.0 for some days without problems.
The update to core 2024.12.0 caused the issue I guess.

**Comment by IMM1906** [comment]
The sema problem after update to core 2024.12.0

**Comment by firebowl** [comment]
Stop posting me too. Subscribe for the issue and wait. Over 20 spam mails in 2 or 3 hours.

**Comment by Shaunpottie** [comment]
Yea I just got another one now because you keep telling people to stop posting so much.

Just unsubscribe and then come back every half hour to see if there is a fix!

Simple solution right?

**Comment by rdilken** [comment]
> I had been using 5.0.0 for some days without problems. The update to core 2024.12.0 caused the issue I guess.

Confirmed. Switched back to 2024.11.3 and it works again.

**Comment by jojkopeter** [comment]
Error after Home Assistant Update:

Setup failed for custom integration 'alexa_media': Requirements for alexa_media not found: ['alexapy==1.29.4'].

**Comment by bkarakashev** [comment]
This is a good workaround that worked for me:
https://github.com/alandtse/alexa_media_player/issues/2698#issuecomment-2508801487

**Comment by Waldi2023** [comment]
Ich habe das zuvor erstellte Backup vom: **core_2024.11.3 wiederhergestellt**, dann funktioniert die Alexa Integration wider einwandfrei. Ich warte nun ab, bis eine fehlerbereinigte Core Version zur Verfügung gestellt wird!

**Comment by user45876** [comment]
The issue is not the core version, but the integration.

**Comment by rahaaas** [comment]
For anyone needing a quick and dirty workaround (after my morning routine failed), it seems 'continue_on_error' won't get past automations failing, as it only supports HA actions. 

So, if you create a building block of 'run in parallel' and run any AMP services in that, you can then nest a 'run in sequence' block below with all the other automation actions in. The first element fails with an error, but the rest of the automation should at least run.

Am sure there's a MUCH better, more tech-savvy approach, but this works, and should (hopefully) future-proof any AMP-falling-over...

**Comment by Petro31** [comment]
Please search before making an issue.

https://github.com/alandtse/alexa_media_player/issues/2698

**Comment by erictroebs** [comment]
> Ich habe das zuvor erstellte Backup vom: **core_2024.11.3 wiederhergestellt**, dann funktioniert die Alexa Integration wieder einwandfrei. Ich warte nun ab, bis eine fehlerbereinigte Core Version: core_2024.12.x zur Verfügung gestellt wird!

This is not a problem with HA core. Instead, `alexapy==1.29.4` requires an older version of `aiofiles` than the new HA version is bundled with. [Manually use `alexapy==1.29.5` until there is an update to this repo.](https://github.com/alandtse/alexa_media_player/issues/2723#issuecomment-2520960635)

**Comment by danielbrunt57** [comment]
> Yea I just got another one now because you keep telling people to stop posting so much.
> Just unsubscribe and then come back every half hour to see if there is a fix!
> Simple solution right?

@Shaunpottie No, that's not a simple solution since as a contributor, I am automatically subscribed to **_everything_**!
And I need to painstaking check each and every email otherwise I may miss one that is _not_ from this issue!
The correct solution is as @firebowl asked, stop posting and use the subscribe button!
Thanks.
